File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/WriteVariableValueResponse.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.builtin.StatusCode;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.HashMap;
import java.util.Map;
import java.util.Set;

public class WriteVariableValueResponse {
    private final Map<UaChildId, StatusCode> results;

    public WriteVariableValueResponse()
    {
        results = new HashMap<>();
    }

    public WriteVariableValueResponse(Map<UaChildId, StatusCode> results) { this.results = results; }

    public void setOperationResults(WriteVariableValueRequest request, StatusCode code)
    {
        Set<UaChildId> childIds = request.getVariableValues().keySet();
        for (UaChildId item : childIds)
        {
            results.put(item, code);
        }
    }

    public void setOperationResult(String id, StatusCode code)
    {
        results.put(new UaChildId(id), code);
    }

    public void setOperationResult(UaChildId id, StatusCode code)
    {
        results.put(id, code);
    }

    public Map<UaChildId, StatusCode> getResults() {
        return results;
    }
}

File: opcua-webservice-node/source/src/server/service/message/reactiveobject/WriteVariableValueResponse.ts
import { UaStatusCode } from "opcua-webapi-ts";
import { UaChildId } from "../../../types";
import { WriteVariableValueRequest } from "./WriteVariableValueRequest";

export class WriteVariableValueResponse {
    private readonly _results: Map<string, UaStatusCode>;

    constructor(results: Map<string, UaStatusCode> = new Map()) {
        this._results = results;
    }

    setOperationResults(request: WriteVariableValueRequest, code: UaStatusCode): void {
        for (const childId of request.variableValues.keys()) {
            this._results.set(childId, code);
        }
    }

    setOperationResult(id: string | UaChildId, code: UaStatusCode): void {
        this._results.set(this.toKey(id), code);
    }

    get results(): Map<string, UaStatusCode> {
        return this._results;
    }

    private toKey(id: string | UaChildId): string {
        if (typeof id === "string") {
            return id;
        }

        return id.serialize();
    }
}